### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by learnwithalfred**
## The following requirements have been completed

Added: 
- A method that returns the 3 most recent posts for a given user.
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
- A method that updates the comments counter for a post.
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -5,4 +5,16 @@ class Post < ApplicationRecord
...
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  after_save :update_posts_counter
+  def update_posts_counter
+    user = self.user
+    user.post_counter = user.posts.count
+    user.save
+  end
```
Comment (DeliceLydia) [inline app/models/post.rb:22]: ## Optional
- [x] After adding Rails call back you can make that method `private`

File: app/models/post.rb [hunk 2]
```diff
@@ -5,4 +5,16 @@ class Post < ApplicationRecord
   validates :text, presence: true, length: { minimum: MIN_LENGTH }
 
   belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+
```
Comment (DeliceLydia) [inline app/models/post.rb]: - [ ] Remember a post can have many `likes` as well please add that association

File: app/models/like.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@ class Like < ApplicationRecord
   after_save :update_likes_count
   def update_likes_count
     post = self.post
-    post.likes_count += 1
+    post.likes_count = post.likes.count
     post.save
   end
```
Comment (DeliceLydia) [inline app/models/like.rb:12]: ## Optional
- [x] After adding Rails call back you can make that method  `private`

Comment (DeliceLydia) [reply app/models/like.rb:12]: - [x] Please add likes associations as well, remember `like`
```ruby
belongs_to :user
belongs_to :post
```

File: app/models/user.rb [hunk 4]
```diff
@@ -6,4 +6,8 @@ class User < ApplicationRecord
   validates :photo, presence: true, length: { minimum: MIN_LENGTH }
   validates :bio, presence: true, length: { minimum: MIN_LENGTH }
   has_many :posts, foreign_key: :author_id
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
```
Comment (DeliceLydia) [inline app/models/user.rb]: - [x] Remember a user can have many `comments` and `likes` on a post please add associations
```ruby
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @learnwithalfred, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/models/post.rb (hunk 1 above)
Comment (learnwithalfred) [reply app/models/post.rb:22]: done

File: app/models/like.rb (hunk 3 above)
Comment (learnwithalfred) [reply app/models/like.rb:12]: It was already added

File: app/models/user.rb (hunk 4 above)
Comment (learnwithalfred) [reply app/models/user.rb]: You are right

**Review by rustam-tolipov — APPROVED** [review]
Hi @learnwithalfred, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
No optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by learnwithalfred** [comment]
Thank you soo mcuh

**Pull request merged**